common-fate/granted
# Add headless mode cfg flag to control auto-login

### What changed?

Instead of prompting the user to log in, log in automatically, unless they specifically opt-out by configuring `HeadlessMode` in the granted config.

### Why?

This was a regression in 0.9.0 for users using `credential-process` on desktop environments, due to removing this behavior for headless users.

### How did you test it?

1. Link up dgranted and dassume, reconfigure my AWS profiles to use those binaries.
2. Purge granted tokens from keychain.
3. Run `aws --profile <profile> iam list-account-aliases`.
  1. It automatically opens my configured browser and logs me in. This confirms that the path for "non-headless" mode works as expected.
4. Purge granted tokens from keychain again.
5. Run `dgranted settings set -s HeadlessMode -v true`
6. Run `aws --profile <profile> iam list-account-aliases`.
  7. It prompts me to run the `granted sso login` command

### Potential risks

This may be initially breaking for headless users until they update configuration

### Is patch release candidate?

No; it should be a minor release because it may be breaking for headless users.

### Link to relevant docs PRs
